### ISSUE TITLE
Random initialization for MLM model

### ISSUE THREAD

**Issue by InhyeokYoo** [john-hewitt/embed-init#1]
Hi.
I read your post and it is really nice. Thank you for the wonderful post.

I tried your experiment to MLM model such as BERT but I don't think BERT suffers the same problem. I think the formula that you derived in the post also apply to BERT.

What do you think about this? Do you think it's just lucky case like you mentioned _"In general, this doesn’t happen for all (or even most) LMs, but when it does, zero-initialization is bad."_ or MLM models doen't suffer this problem?

**Comment by john-hewitt** [comment]
Hi!

Yep, most models do not suffer from this problem. It only happens if, for some inputs, the logits for all words are all less than 0 (e.g., all `-4` or less). The softmax is invariant to any additive constant (e.g., adding -100 to all logits) so it seems to happen sort of accidentally sometimes as a result of... something about the accidental process of optimization? It's an open question to me why it only happens sometimes. I wonder if it has to do with some details of some nonlinearities... maybe differences in robustness to small additive noise in the logit space... really not sure.

The formula I used does apply to BERT, as you say.

I didn't pursue this further than a blog post because it just doesn't seem to be an issue that frequently. Best of luck!

**Comment by InhyeokYoo** [comment]
I appreciate your answer. It helps a lot. I wish you the best of luck!
